test/shutdown_SUITE.erl

```diff
@@ -297,6 +297,54 @@ http10_connection_close(Config) ->
 	{ok, _} = gun:await_body(ConnPid, StreamRef),
 	gun_is_down(ConnPid, ConnRef, normal).
 
+http1_response_connection_close_delayed_body(_) ->
+	doc("HTTP/1.1: Confirm that requests initiated when Gun has received a "
+		"connection: close response header fail immediately if retry "
+		"is disabled, without waiting for the response body."),
+	ServerFun = fun(_Parent, ClientSocket, gen_tcp) ->
+		try
+			{ok, Req} = gen_tcp:recv(ClientSocket, 0, 5000),
+			<<"GET / HTTP/1.1\r\n", _/binary>> = Req,
+			ok = gen_tcp:send(ClientSocket, <<"HTTP/1.1 200 OK\r\n"
+				"Connection: close\r\n"
+				"Content-Length: 12\r\n\r\nHello">>),
+			timer:sleep(500),
+			ok = gen_tcp:send(ClientSocket, " world!")
+		after
+			gen_tcp:close(ClientSocket)
+		end
+	end,
+	{ok, ServerPid, OriginPort} = gun_test:init_origin(tcp, http, ServerFun),
+
+	%% client connects
+	{ok, ConnPid} = gun:open("localhost", OriginPort, #{
+		protocols => [http],
+		retry => 0
+	}),
+	{ok, _Protocol} = gun:await_up(ConnPid),
+	receive {ServerPid, handshake_completed} -> ok end,
+	ConnRef = monitor(process, ConnPid),
+
+	StreamRef1 = gun:get(ConnPid, "/"),
+	StreamRef2 = gun:get(ConnPid, "/"),
+
+	%% We get the response headers with connection: close.
+	{response, nofin, 200, _} = gun:await(ConnPid, StreamRef1),
+
+	%% Pipelined request fails immediately.
+	{gun_error, ConnPid, StreamRef2, closing} = receive E2 -> E2 end,
+
+	{gun_data, ConnPid, StreamRef1, nofin, <<"Hello">>} =
+		receive PartialBody -> PartialBody end,
+
+	%% Request initiated when Gun is in closing state fails immediately.
+	StreamRef3 = gun:get(ConnPid, "/"),
+	{gun_error, ConnPid, StreamRef3, closing} = receive E3 -> E3 end,
+
```

Comment:
Please remove all these empty lines while you're at it.

Reply:
Done.